### PR TITLE
DDF-2726: add alphaNumeric Sort field to solr schema.xml

### PR DESCRIPTION
**Pull request by kcwire**
#### What does this PR do?
fixes sorting on textual fields with multiple words.
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged)?
@mcalcote 
@brianfelix 
#### Select at least one member from relevant component team(s) from below (at least one component team member needs to approve the PR).
[Solr](https://github.com/orgs/codice/teams/solr)
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@kcwire
@pklinef
#### How should this be tested? (List steps with links to updated documentation)
Ingest several files with titles of varying length and multiple words.  Post the following query to the CSW Endpoint and verify the results are sorted alphabetically.

```
<csw:GetRecords xmlns:csw="http://www.opengis.net/cat/csw/2.0.2" xmlns:ogc="http://www.opengis.net/ogc" xmlns:gml="http://www.opengis.net/gml" xmlns:ns4="http://www.w3.org/1999/xlink" xmlns:ns5="http://www.w3.org/2001/SMIL20/" xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:ows="http://www.opengis.net/ows" xmlns:dct="http://purl.org/dc/terms/" xmlns:ns9="http://www.w3.org/2001/SMIL20/Language" resultType="results" outputFormat="application/xml" outputSchema="http://www.opengis.net/cat/csw/2.0.2" startPosition="1" maxRecords="5" service="CSW" version="2.0.2">
  <csw:Query typeNames="csw:Record">
    <csw:ElementSetName>brief</csw:ElementSetName>
    <csw:Constraint version="1.1.0">
      <ogc:Filter>
          <ogc:PropertyIsLike wildCard="*" singleChar="#" escapeChar="!">
            <ogc:PropertyName>title</ogc:PropertyName>
            <ogc:Literal>*</ogc:Literal>
          </ogc:PropertyIsLike>
      </ogc:Filter>
    </csw:Constraint>
    <ogc:SortBy>
      <ogc:SortProperty>
        <ogc:PropertyName>title</ogc:PropertyName>
        <ogc:SortOrder>ASC</ogc:SortOrder>
      </ogc:SortProperty>
    </ogc:SortBy>
  </csw:Query>
</csw:GetRecords>
```

#### What are the relevant tickets?
[DDF-2726](https://codice.atlassian.net/browse/DDF-2726)
#### Checklist:
- [ ] Documentation Updated
- [ ] Change Log Updated
- [x] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: catalog/core/catalog-core-solr/src/test/java/ddf/catalog/source/solr/SolrProviderTest.java [hunk 1]
```diff
@@ -4241,6 +4241,350 @@ public void testSortById() throws Exception {
...
+                        .toDate());
+
+        query = new QueryImpl(filter);
+
+        query.setSortBy(new ddf.catalog.filter.impl.SortByImpl(Metacard.TITLE,
+                SortOrder.ASCENDING.name()));
+
+        sourceResponse = provider.query(new QueryRequestImpl(query));
+
+        assertEquals(list.size(),
```
Comment (bdeining) [inline catalog/core/catalog-core-solr/src/test/java/ddf/catalog/source/solr/SolrProviderTest.java]: We should be using `assertThat` here (and `hasSize`).  https://codice.atlassian.net/wiki/display/DDF/Unit+Testing

File: catalog/core/catalog-core-solr/src/test/java/ddf/catalog/source/solr/SolrProviderTest.java [hunk 2]
```diff
@@ -4241,6 +4241,350 @@ public void testSortById() throws Exception {
...
+        query.setSortBy(new ddf.catalog.filter.impl.SortByImpl(Metacard.TITLE,
+                SortOrder.DESCENDING.name()));
+
+        sourceResponse = provider.query(new QueryRequestImpl(query));
+
+        assertEquals(list.size(),
+                sourceResponse.getResults()
+                        .size());
+
+        int asciiE = 69;
```
Comment (pklinef) [inline catalog/core/catalog-core-solr/src/test/java/ddf/catalog/source/solr/SolrProviderTest.java]: Why was this named `asciiE`?  Does the `E` stand for something?

File: platform/solr/platform-solr-server-standalone/src/main/resources/solr/conf/schema.xml [hunk 3]
```diff
@@ -561,6 +561,25 @@
     </fieldType>
     -->
 
+    <fieldType name="alphaNumericSort" class="solr.TextField" sortMissingLast="true" omitNorms="true">
+        <analyzer>
+            <!-- The TrimFilter removes any leading or trailing whitespace -->
+            <filter class="solr.TrimFilterFactory" />
+            <!-- Truncate the field to 1000 Characters -->
```
Comment (pklinef) [inline platform/solr/platform-solr-server-standalone/src/main/resources/solr/conf/schema.xml]: Update comment to 127 characters.

Comment (kcwire) [inline platform/solr/platform-solr-server-standalone/src/main/resources/solr/conf/schema.xml]: fixed.

File: catalog/core/catalog-core-solr/src/test/java/ddf/catalog/source/solr/SolrProviderTest.java (hunk 2 above)
Comment (kcwire) [inline catalog/core/catalog-core-solr/src/test/java/ddf/catalog/source/solr/SolrProviderTest.java]: 69 == E in ascii.

**Comment by cruzj6** [comment]
Manually tested this, and verified it is working correctly.

**Pull request merged**